src/main/java/nl/vpro/magnolia/module/keycloak/util/MagnoliaReturnToRequestWrapper.java

```diff
@@ -0,0 +1,26 @@
+package nl.vpro.magnolia.module.keycloak.util;
+
+import info.magnolia.cms.security.auth.login.FormLogin;
+
+import java.net.URLEncoder;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class MagnoliaReturnToRequestWrapper extends HttpServletRequestWrapper {
+
+    public MagnoliaReturnToRequestWrapper(HttpServletRequest request) {
+        super(request);
+    }
+
+    @Override
+    public String getQueryString() {
+        String q = super.getQueryString();
+        if (q == null || q.isEmpty()) {
```

Comment:
Wat dacht je van StringUtils.isBlank()? Daar vang je nog wat meer ellende mee af

Reply:
Is een optie.